pkg/scheduler/register.go

```diff
@@ -0,0 +1,109 @@
+package scheduler
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	"github.com/gorilla/mux"
+	"github.com/openshift-pipelines/pipelines-as-code/pkg/apis/pipelinesascode/v1alpha1"
+	"github.com/tektoncd/pipeline/pkg/apis/pipeline/v1beta1"
+	"go.uber.org/zap"
+	"k8s.io/apimachinery/pkg/api/errors"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func (s *scheduler) Register() http.HandlerFunc {
+	return http.HandlerFunc(func(response http.ResponseWriter, request *http.Request) {
+		pipelineRun, repository, validationErr := s.validateRequest(response, request)
+		if validationErr != nil {
+			return
+		}
+
+		err := s.syncManager.Register(pipelineRun, repository, s.pipelineClient)
+		if err != nil {
+			s.logger.Error("failed to register pipelinerun", err)
+			responseWriter(s.logger, http.StatusInternalServerError, "failed to register Pipelinerun", response)
+			return
+		}
+
+		responseWriter(s.logger, http.StatusOK, "request registered successfully!", response)
+	})
+}
+
+func (s *scheduler) validateRequest(response http.ResponseWriter, request *http.Request) (*v1beta1.PipelineRun, *v1alpha1.Repository, error) {
+	vars := mux.Vars(request)
+	prNamespace := vars["namespace"]
+	prName := vars["name"]
```

Comment:
what happens if these are absent?

Reply:
weird that golangci-lint is not catching it 🤔

Reply:
those are `path` params
handler won't be invoked if those are missing